core/adapters/http.go

```diff
@@ -11,19 +11,40 @@ import (
 	"net/url"
 	"path"
 	"strings"
+	"time"
 
+	"chainlink/core/logger"
 	"chainlink/core/store"
 	"chainlink/core/store/models"
 	"chainlink/core/utils"
 )
 
+const defaultTimeout = 5000000000 // 5s
+
+func ParseTimeout(timeoutS string) time.Duration {
+	if timeoutS == "" {
+		return defaultTimeout
+	}
+	timeout, err := time.ParseDuration(timeoutS)
+	if err != nil {
+		logger.Warnf("got invalid value for timeout: %s", timeoutS)
+		return defaultTimeout
+	}
+	return timeout
+}
+
 // HTTPGet requires a URL which is used for a GET request when the adapter is called.
 type HTTPGet struct {
 	URL          models.WebURL   `json:"url"`
 	GET          models.WebURL   `json:"get"`
 	Headers      http.Header     `json:"headers"`
 	QueryParams  QueryParameters `json:"queryParams"`
 	ExtendedPath ExtendedPath    `json:"extPath"`
+	timeout      string          `json:"timeout,omitempty"`
```

Comment:
I don't think non-exported struct fields can be decoded from JSON (unless you write a manual decoder).  Have you checked to make sure the field gets correctly populated by a job spec?  I'd probably err on the side of capitalizing/exporting it just in case.

Reply:
No, but this is an excellent point. I will write a test for it.